include/chemfiles/formats/PDB.hpp

```diff
@@ -41,21 +43,31 @@ class PDBFormat final: public Format {
     void read_CRYST1(Frame& frame, const std::string& line);
     // Read ATOM and HETATM records
     void read_ATOM(Frame& frame, const std::string& line, bool is_hetatm);
-
+    // Read secondary structure records. All push secinfo_ vector if line is valid
+    void read_HELIX(const std::string& line);
+    // reads SHEET and TURN records. i1 and i2 are the indicies of the chain ids
+    void read_secondary(const std::string& line, size_t i1, size_t i2,
+                        const std::string& sec);
     // Read CONECT record
     void read_CONECT(Frame& frame, const std::string& line);
 
     std::unique_ptr<TextFile> file_;
-    /// Map of residues, indexed by residue id.
-    std::map<size_t, Residue> residues_;
+    /// Map of residues, indexed by residue id and chainid.
+    std::map<std::pair<char, size_t>, Residue> residues_;
     /// Storing the positions of all the steps in the file, so that we can
     /// just `seekg` them instead of reading the whole step.
     std::vector<std::streampos> steps_positions_;
     /// Number of models written/read to the file.
     size_t models_;
+    /// List of all atom offsets. This maybe pushed in read_ATOM or if a TER
+    /// record is found. It is reset every time a frame is read.
+    std::vector<size_t> atom_offsets_;
     /// Did we wrote a frame to the file? This is used to check wheter we need
     /// to write a final `END` record in the destructor
     bool written_ = false;
+    /// Store secondary structure information. First field is the chainid,
+    /// followed by the first and last residue id in the secondary structure.
+    std::vector<std::tuple<char, size_t, size_t, std::string>> secinfo_;
```

Comment:
This could use a `struct secinfo_meta {char chainid; size_t first; size_t last}` instead of the tuple for more readability. But I'm fine with the std::tuple too.

Reply:
I don't like making structures that are effectively single use. Since this is single use (only used in `PDB` and only to read secondary structure) I think a Tuple is better.

Reply:
I find a struct with names more readable that `std::get<0>`, but let's agree to disagree here =)